src/lib.rs

```diff
@@ -312,35 +312,53 @@ where
     /// based on data ready interrupt or if reading in a tight loop you should
     /// waiting for `is_data_ready`
     fn accel_norm(&mut self) -> Result<F32x3, AccelerometerError<Self::Error>> {
-        let acc_raw: I16x3 = self.accel_raw()?;
-        let sensitivity = match self.get_range()? {
-            Range::G16 => 0.012,
-            Range::G8 => 0.004,
-            Range::G4 => 0.002,
-            Range::G2 => 0.001,
+        // The official driver from ST was used as a reference.
+        // https://github.com/STMicroelectronics/STMems_Standard_C_drivers/tree/master/lis3dh_STdC
+        let mode = self.get_mode()?;
+        let range = self.get_range()?;
+
+        // See "2.1 Mechanical characteristics" in the datasheet to find the
+        // values below. Scale values have all been divided by 1000 in order
+        // to convert the resulting values from mG to G, while avoiding doing
+        // any actual division on the hardware.
+        let scale = match (mode, range) {
+            // High Resolution mode
+            (Mode::HighResolution, Range::G2) => 0.001,
+            (Mode::HighResolution, Range::G4) => 0.002,
+            (Mode::HighResolution, Range::G8) => 0.004,
+            (Mode::HighResolution, Range::G16) => 0.012,
+            // Normal mode
+            (Mode::Normal, Range::G2) => 0.004,
+            (Mode::Normal, Range::G4) => 0.008,
+            (Mode::Normal, Range::G8) => 0.016,
+            (Mode::Normal, Range::G16) => 0.048,
+            // Low Power mode
+            (Mode::LowPower, Range::G2) => 0.016,
+            (Mode::LowPower, Range::G4) => 0.032,
+            (Mode::LowPower, Range::G8) => 0.064,
+            (Mode::LowPower, Range::G16) => 0.192,
         };
 
-        Ok(F32x3::new(
-            (acc_raw.x >> 4) as f32 * sensitivity,
-            (acc_raw.y >> 4) as f32 * sensitivity,
-            (acc_raw.z >> 4) as f32 * sensitivity,
-        ))
+        // Depending on which Mode we are operating in, the data has different
+        // resolution. Using this knowledge, we determine how many bits the
+        // data needs to be shifted.
+        let shift: u8 = match mode {
+            Mode::HighResolution => 4, // High Resolution:  12-bit
+            Mode::Normal => 6,         // Normal:           10-bit
+            Mode::LowPower => 8,       // Low Power:         8-bit
+        };
+
+        let acc_raw = self.accel_raw()?;
+        let x = (acc_raw.x >> shift) as f32 * scale;
```

Comment:
Id have to do the math again, but I think if we just do the scale we dont need the shift? My thought was being cute and trying to accomplish some of those scales with shifts for performance reasons.

Reply:
I'm still a bit shaky on this unfortunately, so hopefully this is correct 😅 

Depending on the operating mode, the data resolution is a different number of bits. The section __3.2.1 High-resolution, normal mode, low-power mode__ in the datasheet (among others) demonstrates this.

The descriptions for the accelerometer data registers also state _"The value is expressed as two’s complement left-justified."_.

So what we're doing with `shift` is right-justifying the data, based on which operating mode is active, and we're using `scale` to perform the actual scaling required.

Reply:
aha.. that may be. This is why we have the buddy system.. ;) 

ill test a bit again.

Reply:
Yeah I think this looks good. I definitely had weirdness in my numbers before. Can you comment that the shift is for left justified etc just like you did above. future me is going to forget this again in the future.

Reply:
Thanks for testing! I have noted the reason for the shift in a comment, but I am going to hold off on committing for now just in case there are further changes to make.